demo-frontend/Main.elm

```diff
@@ -0,0 +1,91 @@
+module Main exposing (..)
+
+import Html.Styled exposing (..)
+import Html.Styled.Attributes exposing (..)
+import Html.Styled.Events exposing (..)
+import Css exposing (..)
+
+import Http
+import File exposing (File)
+import File.Select as Select
+import Json.Encode as Encode exposing (..)
+import Json.Decode as Decode exposing (..)
+import Browser
+import Task
+
+import Components exposing (..)
+import Utils exposing (..)
+import Types exposing (..)
+
+-- TODO: add displaying the palette
+
+endpointUrl : String
+endpointUrl = "http://localhost:3000/upload"
+
+
+
+
+apiRequestEncoder : Algorithm -> ImageString -> Encode.Value
+apiRequestEncoder algorithm image = Encode.object [
+    ("algorithm", Encode.string (algorithmToString algorithm)),
+    ("image", Encode.string image)
+    ]
+
+paletteDecoder : Decode.Decoder (List Types.Color)
+paletteDecoder = Decode.list Decode.string
+
+
+getPaletteFromImage : Algorithm -> ImageString -> Cmd Msg
+getPaletteFromImage algorithm image = Http.post {
+    url = endpointUrl,
+    body = Http.jsonBody (apiRequestEncoder algorithm image),
+    expect = Http.expectJson GotPalette paletteDecoder
+    }
+
+initialState : State
+initialState = {
+    algorithm = Nothing,
+    image = Nothing,
+    colors = [],
+    errorMessage = Nothing
+    }
+
+-- main functions
```

Comment:
proponuję przenazwić może te funkcje, by było wiadomo czego dotyczą operacje (rozumiem że głównego ekranu czy czegoś takiego?) np. `mainView`, `mainUpdate`

Reply:
Usunąłem komentarz, bo on został z czasów jak robiłem prototyp w jednym pliku. Co do nazw, to jak tylko przeczytasz wprowadzenie na ich stronie, to zobaczysz, że view i update to konwencja ich architektury i każda aplikacja w elmie ma te 2 funkcje, więc wiadomo ocb.